clusterloader2/pkg/measurement/common/simple/wait_for_pods.go

```diff
@@ -74,39 +77,51 @@ func (*waitForRunningPodsMeasurement) Execute(config *measurement.MeasurementCon
 	time.AfterFunc(timeout, func() {
 		close(stopCh)
 	})
-	return summaries, waitForPods(config.ClientSet, namespace, labelSelector, fieldSelector, desiredPodCount, stopCh, true)
+	return summaries, waitForPods(config.ClientSet, namespace, labelSelector, fieldSelector, desiredPodCount, stopCh, true, waitForRunningPodsMeasurementName)
 }
 
 // Dispose cleans up after the measurement.
 func (*waitForRunningPodsMeasurement) Dispose() {}
 
-func waitForPods(clientSet clientset.Interface, namespace, labelSelector, fieldSelector string, desiredPodCount int, stopCh <-chan struct{}, log bool) error {
+func waitForPods(clientSet clientset.Interface, namespace, labelSelector, fieldSelector string, desiredPodCount int, stopCh <-chan struct{}, log bool, callerName string) error {
 	// TODO(#269): Change to shared podStore.
 	ps, err := measurementutil.NewPodStore(clientSet, namespace, labelSelector, fieldSelector)
 	if err != nil {
 		return fmt.Errorf("pod store creation error: %v", err)
 	}
 	defer ps.Stop()
 
-	var runningPodsCount int
+	var podsStatus measurementutil.PodsStartupStatus
+	var initialized bool
+	var oldPods []*corev1.Pod
 	for {
 		select {
 		case <-stopCh:
-			return fmt.Errorf("timeout while waiting for %d pods to be running in namespace '%v' with labels '%v' and fields '%v' - only %d found running", desiredPodCount, namespace, labelSelector, fieldSelector, runningPodsCount)
+			return fmt.Errorf("timeout while waiting for %d pods to be running in namespace '%v' with labels '%v' and fields '%v' - only %d found running", desiredPodCount, namespace, labelSelector, fieldSelector, podsStatus.Running)
 		case <-time.After(defaultWaitForPodsInterval):
 			pods := ps.List()
-			runningPodsCount = 0
-			for i := range pods {
-				if pods[i].Status.Phase == corev1.PodRunning {
-					runningPodsCount++
+			podsStatus = measurementutil.ComputePodsStartupStatus(pods, desiredPodCount)
+			if initialized {
+				diff := measurementutil.DiffPods(oldPods, pods)
+				deletedPods := diff.DeletedPods()
+				if len(oldPods) <= desiredPodCount && len(deletedPods) > 0 {
```

Comment:
I'm not sure this condition is what we want.
Because then the following sequence is fine:
0 -> 5 -> 10 -> 5 (desired 5)

When initializing you should mark if this is expected to be increasing or decreasing and do checking based on that.

Reply:
Done.